packages/ingest/PdrFetcher.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const aws = require('@cumulus/common/aws');
+const fs = require('fs-extra');
+const get = require('lodash.get');
+const log = require('@cumulus/common/log');
+const os = require('os');
+const path = require('path');
+const { CollectionConfigStore } = require('@cumulus/common');
+const { parsePdr } = require('./parse-pdr');
+const { buildProviderClient } = require('./providerClientUtils');
+
+const upload = async (bucket, key, filename, tempFile) => {
+  let fullKey = path.join(key, filename);
+
+  // handle the edge case where leading / in key creates incorrect path
+  // by remove the first slash if it exists
+  if (fullKey[0] === '/') {
+    fullKey = fullKey.substr(1);
```

Comment:
Suggest this be replaced with something like `fullkey.replace(/^\/+/, '')` to sanitize input for multiple leading slashes.